### PR TITLE
Make setup when devicemapper devices exist succeed

### PR DESCRIPTION
**Pull request by mulkieran**
If the metadata filesystem is already mounted do not remount it.

### CODE REVIEW

**Comment by mulkieran** [comment]
If the test does not fail, we have some investigating to do.

**Comment by mulkieran** [comment]
Yes, the test fails.

**Comment by mulkieran** [comment]
With old devicemapper:
https://travis-ci.org/stratis-storage/stratisd/jobs/259396315
thread 'loop_test_pool_setup' panicked at 'called `Result::unwrap()` on an `Err` value: DM(Io(Error { repr: Os { code: 16, message: "Device or resource busy" } }))', /checkout/src/libcore/result.rs:860

With new devicemapper:
https://travis-ci.org/stratis-storage/stratisd/jobs/259375804
thread 'loop_test_pool_setup' panicked at 'called `Result::unwrap()` on an `Err` value: Nix(Sys(EBUSY))', /checkout/src/libcore/result.rs:860

**Comment by mulkieran** [comment]
Things are busy, apparently. It is rational to conclude that the new error arises because the device that is already there is busy for some reason, simply from a look at the changes in behaviour introduced in devicemapper.

**Comment by mulkieran** [comment]
Sleeping for half a second makes no difference.

**Comment by mulkieran** [comment]
The good news is that we're failing while setting up the metadata volume which means that likely the devicemapper changes are correct.

**Comment by mulkieran** [comment]
We're failing precisely when we try to mount the mdv filesystem.

**Comment by mulkieran** [comment]
Depend on new devicemapper PR.

**Comment by mulkieran** [comment]
@agrover This is obviously preliminary, because we would expect that the new version of stratisd would pull from the new devicemapper release. There are also some oddities in the test framework changes for faster feedback which I wouldn't commit.

Feel free to take a look, and, in particular, if you have any suggestions for better testing that would be helpful.

**Comment by agrover** [comment]
Looks good to me (minus the just-for-development stuff). The one thought on testing I had was testing a *partially* existing stack of dm devices, but thinking about it more, it would be hard to construct and wouldn't tell us much more than the full setup-pre-existing case.

**Comment by mulkieran** [comment]
Now using devicemapper master branch.

**Comment by mulkieran** [comment]
Now using stratisd devicemapper master branch.

**Comment by mulkieran** [comment]
This is ready to go, except that the testing commit must be removed before final merge.

@agrover @trgill review?

File: tests/util/setup_tests.rs [hunk 1]
```diff
@@ -176,3 +186,45 @@ pub fn test_pool_rename(paths: &[&Path]) {
     assert_eq!(pool_name, name2);
     engine.teardown().unwrap();
 }
+
+/// Verify that setting up a pool when the pool has not been previously torn
+/// down does not fail. Clutter the orignally set up pool with a filesystem
```
Comment (agrover) [inline tests/util/setup_tests.rs]: Looks very good to me, especially this test. 👍 from me once temp stuff is removed, and fix a typo in the above comment line.

**Comment by mulkieran** [comment]
Final rebase. Should be ready to go.

**Pull request merged**